### PR TITLE
Fix #57

### LINKED ISSUE #57: "collapseAllSections()" displays all sections regardless of what shouldShowHeadersForEmptySections says
- [x] I have verified there are no duplicate active or recent bugs, questions, or requests
- [x] I have verified that I am using the latest version of this library.
- [x] I have given my issue a non-generic title.

###### Include the following:
 - Library version: `0.5.0`

###### Reproduction Steps

1.  Use the sample app. Call collapseAllSections().  Section 1 will be rendered even if it's an empty section.

###### Expected Result
Only non-empty section titles are displayed when shouldShowHeadersForEmptySections is false.
###### Actual Result
All section titles were displayed.

screen shot 2017-08-17 at 1 43 56 pm
screen shot 2017-08-17 at 1 44 25 pm

### PR DESCRIPTION
**Pull request by boswelja**
Also clean up some warnings and update gradle
Full list of changes can be found in the commit descriptions

### CODE REVIEW

File: library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java [hunk 1]
```diff
@@ -246,19 +246,16 @@ public final int getItemViewType(int position) {
     }
   }
 
-  @IntRange(from = 0, to = Integer.MAX_VALUE)
```
Comment (afollestad) [inline library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java]: These int range checks are intentional

File: sample/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -13,7 +13,7 @@
             android:label="@string/app_name">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
-
+                <action android:name="android.intent.action.VIEW"/>
```
Comment (afollestad) [inline sample/src/main/AndroidManifest.xml]: This added action isn't needed

File: library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java (hunk 1 above)
Comment (boswelja) [reply library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java]: Oh? It returns a value from outside the range though?

File: sample/src/main/AndroidManifest.xml (hunk 2 above)
Comment (boswelja) [reply sample/src/main/AndroidManifest.xml]: No, but it looks better than suppressing a warning

File: library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java (hunk 1 above)
Comment (afollestad) [reply library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java]: @boswelja that's what the comment is for, it ignores the issue within the library. It's for overriders, so implementers don't accidentally return an internal value.

Comment (boswelja) [reply library/src/main/java/com/afollestad/sectionedrecyclerview/SectionedRecyclerViewAdapter.java]: Fixed it

File: sample/src/main/AndroidManifest.xml (hunk 2 above)
Comment (boswelja) [reply sample/src/main/AndroidManifest.xml]: Will replace it with a warning suppression if you think that will be better

Comment (afollestad) [reply sample/src/main/AndroidManifest.xml]: Well, it's not unnecessary to suppress all warnings

**Pull request closed without merging**